plugins/experimental/esi/lib/Variables.cc

```diff
@@ -357,9 +357,26 @@ Variables::_parseCookieString(const char *str, int str_len)
   AttributeList cookies;
   Utils::parseAttributes(str, str_len, cookies, ";,");
   for (AttributeList::iterator iter = cookies.begin(); iter != cookies.end(); ++iter) {
-    _insert(_dict_data[HTTP_COOKIE], string(iter->name, iter->name_len), string(iter->value, iter->value_len));
-    _debugLog(_debug_tag, "[%s] Inserted cookie with name [%.*s] and value [%.*s]", __FUNCTION__, iter->name_len, iter->name,
-              iter->value_len, iter->value);
+    std::string v = iter->name;
+    size_t eq     = v.find("=");
+
+    if (eq != std::string::npos) {
+      v = v.substr(0, eq);
+    }
+
+    bool found = false;
+    for (Utils::HeaderValueList::iterator lz = _whitelistCookies.begin(); lz != _whitelistCookies.end(); ++lz) {
+      std::string c = *lz;
```

Comment:
Maybe this is the norm for the esi, but why create this intermediary std::string c? That seems inefficient and unnecessary ? Can't you just use the *lz ?